### ISSUE TITLE
♻️ Remove underline from the accordion component title

### ISSUE THREAD

**Issue by substrae** [GCTC-NTGC/gc-digital-talent#8028]
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/817f3e71-bb1f-48b3-bfbb-afe312ad7d2a)

## ✨ Summary

[Visit the Figma file](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=5-20&mode=design) • Chat with @substrae for design questions

We've received a handful of reports that users are hesitant to click accordions because the underline makes them think it will take them to a new page. As such, we're going to try removing it to see if the chevron is enough of a signifier on its own.

```[tasklist]
### ✅ Requirements
- [ ] Remove the underline from accordion titles/headings
- [ ] Remove the `<hr>` or border between the accordion title and content when expanded
```

**Comment by esizer** [comment]
Could this just be added to the AC of the [accordion rework](https://github.com/GCTC-NTGC/gc-digital-talent/issues/6734) to prevent it from lagging behind?

**Comment by substrae** [comment]
@esizer Yeah, sure thing. I created this ticket assuming the previous work was done already.

**Comment by esizer** [comment]
Closing this as fixed by #8402
